content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+#### Request Fields
+
+A summary of fields relevant to authorization requests with the atproto OAuth profile:
+
+- `client_id` (string, required): identifies the client software. See "Clients" section above for details.
+- `response_type` (string, required): must be `code`
+- `code_challenge` (string, required): the PKCE challenge value. See "PKCE" section.
+- `code_challenge_method` (string, required): which code challenge method is used, for example `S256`. See "PKCE" section.
+- `state` (string, required): random token used to verify the authorization request against the response. See below.
+- `redirect_uri` (string, required): must match against URIs declared in client metadata and have a format consistent with the `application_type` declared in the client metadata. See below.
+- `scope` (string with space-separated values, required): must be a subset of the scopes declared in client metadata. Must include `atproto`. See "Scopes" section.
+- `client_assertion_type` (string, optional): used by confidential clients to describe the client authentication mechanism. See “Confidential Client” section.
+- `client_assertion` (string, optional): only used for confidential clients, for client authentication. See "Confidential Client" section.
+- `login_hint` (string, optional): account identifier to be used for login. See "Authorization Interface" section.
+
+The `client_secret` value, used in many other OAuth profiles, should not be included.
+
+The `state` parameter in client authorization requests is mandatory. Clients should use randomly-generated tokens for this parameter and not have collisions or reuse tokens across any combination of device, account, or session. Authorization Servers should reject duplicate state parameters, but are not currently required to track state values across accounts or sessions.
+
+For web clients, the `redirect_uri` is a HTTPS URL which will be redirected in the browser to return users to the application at the end of the Authorization flow. The `redirect_uri` must match one of the URIs declared in the client metadata and the Authorization Server must verify this condition. There is a special exception for the [localhost](http://localhost) development workflow to use `http://`, as described in the "Localhost Client Development" section. For native clients, the `redirect_uri` may use a custom URI scheme to have the operating system redirect the user back to the app, instead of a web browser. The custom scheme must match the `client_id` hostname in reverse-domain order. The URI scheme must be followed by a single colon (`:`) then a single forward slash (`/`) and then a URI path component. For example, an app with `client_id`  [`https://app.example.com/client-metadata.json`](https://app.example.com/client-metadata.json) could have a `redirect_uri` of `com.example.app:/callback`.
```

Comment:
The redirect URI *must* be defined, and excaclty match, one of the redirect uri defined in the metadata document.

Exact matching is required, except for `http://127.0.0.1` and `http://[::1]` redirect uris (used by `native` apps) where the port number must not be specified in the document metadata, and can be any value (excluded from the matching) during the auth request.

Reply:
I think the text already covers the first part?

For the second, I don't want to re-describe all the matching rules here, I think it should stick in the localhost section. Added some text there.